homeassistant/components/qnap/sensor.py

```diff
@@ -201,18 +196,16 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the QNAP NAS sensor."""
-    api = QNAPStatsAPI(config)
-    api.update()
-
-    # QNAP is not available
-    if not api.data:
+    coordinator = QnapCoordinator(hass, config)
+    await coordinator.async_refresh()
```

Comment:
In the config flow PR you will want to replace this by `await coordinator.async_config_entry_first_refresh()`